### PR TITLE
Add command component to developer guide and rectify other errors

### LINKED ISSUE #385: Furnish command command component to developer guide
Add class diagram
Add description
Add general workflow

### PR DESCRIPTION
**Pull request by warrencxw**
Fix sequence diagram of SessionSummaryCommand and Parser Component
Add colour to sequence diagram entities
Add Command component class diagram
Add Command component description
Add Command component workflow

Fixes #385

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -116,13 +119,38 @@ The general workflow of the `Parser` component is as follows:
...
+   * Each of these parse methods in `Parser` class then calls utility methods from `ParserUtils` class 
+     to return a parsed value.
 4. All relevant arguments that are parsed will then be used to create a new `XYZCommand `object which is
    then returned to the `Parser` class.
 5. The created `XYZCommand` object is then returned to the `SplitLah` object to be run.
 
 ### Command Component
+![Command Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/CommandComponent.drawio.png)
+<br>
+The `Command` component consist of the Command class and all the subclasses of the Command class
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:131]: Possible to change to consists

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -116,13 +119,38 @@ The general workflow of the `Parser` component is as follows:
...
+(also referred to as XYZCommand collectively).
+* The subclasses include _data related commands_ such as `SessionJKLCommand`, `ActivityJKLCommand` and `GroupJKLCommand`,
+  each representing commands that work with `Session`, `Activity` and `Group` classes respectively
+  (JKL here refers to either `Create`, `Delete`, `Edit`, `List`, `View` or `Summary`).
+* The subclasses also include _utility commands_ such as `HelpCommand`, `ExitCommand` and `InvalidCommand`
+  (hereafter collectively referred to as `UtilityCommand`).
+
+The general workflow of the `Command` component is as follows:
+1. After a `XYZCommand` object is created by the [`Parser` Component](#parser-component),
+   it is passed back to the `SplitLah` object to be run.
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:141]: Possible to remove "be"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -101,12 +101,15 @@ As TextUI handles all input and output streams, these streams can be changed wit
 ### Parser Component
 ![Parser Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ParserComponent.drawio.png)
 <br>
-The `Parser` component consists of the `Parser` class, `ParserUtils` class as well as the `ParserErrors` class.<br>
-The `Parser` class provides utility methods to parse commands and arguments from the user and
-return a `Command` object representing an instruction that the user has for SplitLah.
-`Parser` class is the only class in the `Parser` component that other external classes interact with.<br>
-On the other hand, the `ParserUtils` class provide supporting methods for `Parser` class to properly run,
-and `ParserErrors` class provide methods to produce custom error messages for the `Parser` component.
+The `Parser` component consists of the `Parser` class, `ParserUtils` class, `ParserErrors` class
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:104]: Possible to add a comma at the end of this line

**Review by ivanaitzliddat — APPROVED** [review]
Looks good to merge, but left some comments for minor grammatical corrections

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -116,13 +119,38 @@ The general workflow of the `Parser` component is as follows:
    (`XYZCommand` is a placeholder for specific subclass of the `Command` class, e.g. `SessionCreateCommand`)
 3. The `XYZCommandParser` object will then use parse methods from `Parser` class to extract all the
    arguments from the user input.
-   1. Each of these parse methods in `Parser` class then calls utility methods from `ParserUtils` class
-      to return a parsed value.
+   * Each of these parse methods in `Parser` class then calls utility methods from `ParserUtils` class 
+     to return a parsed value.
 4. All relevant arguments that are parsed will then be used to create a new `XYZCommand `object which is
    then returned to the `Parser` class.
 5. The created `XYZCommand` object is then returned to the `SplitLah` object to be run.
```
Comment (matheril) [inline docs/DeveloperGuide.md:126]: Consider changing these steps to present tense, as currently it switches between present and future.

e.g. "1. When required to parse for a command, the running `SplitLah` object passes a String object containing user input to the `Parser` class."

Comment (matheril) [reply docs/DeveloperGuide.md:126]: Past tense is okay when referring to a previous step. e.g.

"4. All relevant arguments that were parsed are then used to create a new `XYZCommand `object which is
   then returned to the `Parser` class.

**Review by matheril — CHANGES_REQUESTED** [review]
Consider changing future tense to present tense.

**Review by matheril — APPROVED** [review]
Looks good to merge

**Review by froststein — APPROVED** [review]
Looks good to merge

**Pull request merged**